lib/Site.js

```diff
@@ -549,9 +549,15 @@ Site.prototype.generatePages = function () {
 Site.prototype.regenerateAffectedPages = function (filePaths) {
   const builtFiles = {};
   const processingFiles = [];
+  const variablePath = path.resolve(this.rootPath, USER_VARIABLES_PATH);
+  const variableFileChanged = filePaths.includes(variablePath);
+  if (variableFileChanged) {
+    this.collectUserDefinedVariablesMap();
+    logger.warn('Rebuilding all pages as variable file was changed');
+  }
```

Comment:
1. Maintain plural `variables`, e.g. `variables file` and `variablesPath`.
1. Can we extract this to:

```js
const shouldRebuildAllPages = this.collectUserDefinedVariablesMapIfNeeded(filePaths);
if (shouldRebuildAllPages) {
  logger.warn('Rebuilding all pages as variables file was changed');
}
```
where:
```js
Site.prototype.collectUserDefinedVariablesMapIfNeeded = function (filePaths) {
  const variablesPath = path.resolve(this.rootPath, USER_VARIABLES_PATH);
  if (filePaths.includes(variablesPath)) {
    this.collectUserDefinedVariablesMap();
    return true;
  }
  return false;
}
```

3. Should short-circuit:
```js
if (shouldRebuildAllPages || filePaths.some(filePath => page.includedFiles[filePath])) {
```